lib/jwt/jwk/rsa.rb

```diff
@@ -51,6 +51,9 @@ def verify_key
       def export(options = {})
         exported = parameters.clone
         exported.reject! { |k, _| RSA_PRIVATE_KEY_ELEMENTS.include? k } unless private? && options[:include_private] == true
+
+        exported[:x5t] = Base64.url_encode(OpenSSL::Digest::SHA1.new(rsa_key.to_der).digest) if options[:include_x5t]
```

Comment:
Im still a bit uncertain if we should make the decision here what the thumbprint is, the x5t is the thumbprint for the certificate, the RSA public key is just a part of the certificate.

Think we could have a way to export a X509 certificate that could be adding these fields directly from the certificate itself.

Also with this approach the thumbprint will differ if rsa_key is a private key or just the public key.
Also what about the SHA256 version of the thumbprint?

Reply:
That's a good point, thanks for pointing that out. Let me think a bit more about this.

Reply:
What if we would for now only focus on the key finding part? so if the JWK has a `x5*` header we are able to map them to the tokens that also have them.

Reply:
I think the key binding should work with the code in https://github.com/jwt/ruby-jwt/pull/669/files#diff-7cb47ecf35db922bbd3b543b5df2e594c4ffc8f6bdd62c34f1b8544d6089e00fR58.

I'll remove this code (currently it's mostly used for testing anyway). What do you think?

Reply:
Got it. Let me update the code.